components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -504,7 +504,7 @@ func SetContainerImageFromRegistry(ctx context.Context, config *rest.Config, not
 							// List imagestreams in the specified namespace
 							imagestreams, err := dynamicClient.Resource(ims).Namespace(namespace).List(ctx, metav1.ListOptions{})
 							if err != nil {
-								log.Error(err, "Cannot list imagestreams", "namespace", namespace)
+								log.Info("Cannot list imagestreams", "namespace", namespace, "error", err)
```

Comment:
Better solution than just this change would be to detect presence of these namespaces first and then do the rest of the stuff. If none of them is present, then something is wrong of course. In a standard situation, only one is present. If both are present, there is some risk we are intervening with some custom namespace in one case. I don't have the answer for the problem when both are present.

Reply:
Actually... why don't we simply check the namespace in which the odh-notebook-controller (this instance) is running? :thinking:

Reply:
I'd

/approve

this as a quick fix for the release, so it can go to branch, and maybe worry about restructuring the logic for the future.

Reply:
I feel we should drop `err` in the log 
with `err` it would be the same as current logic
it would print the whole stack trace, that sa is forbidden.

Reply:
Does it really print the whole stacktrace? The following program at https://go.dev/play/p/qkxdV0aVyF7 only prints the one-line error message, when I do

```go
	_, err := os.Open("no-such-file")
	log.Info("Cannot list imagestreams", "namespace", "someNamespace", "error", err)
```

Reply:
Right, tested on a cluster without opendatahub namespace, and we got the log without the stacktrace:

```
{"level":"info","ts":"2024-06-07T08:47:53Z","logger":"controllers.Notebook","msg":"Cannot list imagestreams","notebook":"w1","namespace":"atheo","namespace":"opendatahub","error":"imagestreams.image.openshift.io is forbidden: User \"system:serviceaccount:redhat-ods-applications:odh-notebook-controller-manager\" cannot list resource \"imagestreams\" in API group \"image.openshift.io\" in the namespace \"opendatahub\""}
```

just for the record before that change the log was like:
```
{"level":"error","ts":"2024-06-07T08:54:01Z","logger":"controllers.Notebook","msg":"Cannot list imagestreams","notebook":"wqw","namespace":"atheo","namespace":"opendatahub","error":"imagestreams.image.openshift.io is forbidden: User \"system:serviceaccount:redhat-ods-applications:odh-notebook-controller-manager\" cannot list resource \"imagestreams\" in API group \"image.openshift.io\" in the namespace \"opendatahub\"","stacktrace":"github.com/opendatahub-io/kubeflow/components/odh-notebook-controller/controllers.SetContainerImageFromRegistry\n\t/opt/rhods/odh-notebook-controller/controllers/notebook_webhook.go:503\ngithub.com/opendatahub-io/kubeflow/components/odh-notebook-controller/controllers.(*NotebookWebhook).Handle\n\t/opt/rhods/odh-notebook-controller/controllers/notebook_webhook.go:260\nsigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).Handle\n\t/remote-source/deps/gomod/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/webhook/admission/webhook.go:169\nsigs.k8s.io/controller-runtim...
```

Reply:
logging "namespace" twice, and a different one each time, is weird, though. my go playground does it, and the log from cluster in previous comment has it too

```
"namespace":"atheo","namespace":"opendatahub"
```

Reply:
> Actually... why don't we simply check the namespace in which the odh-notebook-controller (this instance) is running? 🤔

I created a followup issue for the https://issues.redhat.com/browse/RHOAIENG-8390

Reply:
Thanks Jan for the seperate issue. 
About the double namespace mention, i modified a bit the log and now it show the bellow message (the `"notebook":"q2","namespace":"rest"` are shown by default of every object when use the info log):
```
{"level":"info","ts":"2024-06-11T10:57:56Z","logger":"controllers.Notebook","msg":"Cannot list imagestreams","notebook":"q2","namespace":"rest","error":"imagestreams.image.openshift.io is forbidden: User \"system:serviceaccount:opendatahub:odh-notebook-controller-manager\" cannot list resource \"imagestreams\" in API group \"image.openshift.io\" in the namespace \"redhat-ods-applications\""}
```